File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/endpoints/pubsub/Publish.java
package com.pubnub.api.java.endpoints.pubsub;

import com.pubnub.api.java.endpoints.Endpoint;
import com.pubnub.api.models.consumer.PNPublishResult;

public interface Publish extends Endpoint<PNPublishResult> {
    Publish message(Object message);

    Publish channel(String channel);

    Publish shouldStore(Boolean shouldStore);

    Publish usePOST(boolean usePOST);

    Publish meta(Object meta);

    Publish replicate(boolean replicate);

    Publish ttl(Integer ttl);

    Publish type(String type); //todo messageType
}

File: pubnub-kotlin/pubnub-kotlin-api/src/commonMain/kotlin/com/pubnub/kmp/endpoints/pubsub/Publish.kt
package com.pubnub.kmp.endpoints.pubsub

import com.pubnub.kmp.PNFuture
import com.pubnub.api.models.consumer.PNPublishResult

/**
 * @see [PubNub.publish]
 */
interface Publish : PNFuture<PNPublishResult>